### DISCUSSION TITLE
local and globle coordinates

### DISCUSSION THREAD

**Discussion by sunmengnan** [Nicholasli1995/EvoSkeleton#51] [Q&A] [answered]
本地坐標係（比如，上半身坐標系），X和Y軸一定是正交的嗎？

**Comment by Nicholasli1995** [comment, marked as answer]
> 本地坐標係（比如，上半身坐標系），X和Y軸一定是正交的嗎？

是正交的，有gram_schmidt 正交化：
https://github.com/Nicholasli1995/EvoSkeleton/blob/b2b355f4c1fa842709f100d931189ce80008f6ef/libs/skeleton/anglelimits.py#L298
